### PR TITLE
Fix invalid joinColumn in ClassMetadata::associationMappings

### PR DESCRIPTION
**Pull request by gadelkareem**
Fix getting `Notice: Undefined index: user_id` for a user field association

### CODE REVIEW

File: lib/Doctrine/ORM/Utility/IdentifierFlattener.php [hunk 1]
```diff
@@ -85,6 +85,9 @@ public function flattenIdentifier(ClassMetadata $class, array $id)
                 }
 
                 $flatId[$field] = implode(' ', $associatedId);
+            } elseif (array_key_exists($field, $id)) {
```
Comment (phansys) [inline lib/Doctrine/ORM/Utility/IdentifierFlattener.php]: Can `$id[$field]` be `null`? If that isn't the case, `isset()` is able to deliver better performance.

File: lib/Doctrine/ORM/Utility/IdentifierFlattener.php [hunk 2]
```diff
@@ -93,8 +96,7 @@ public function flattenIdentifier(ClassMetadata $class, array $id)
                 }
 
                 $flatId[$field] = implode(' ', $associatedId);
-            } else {
-                $flatId[$field] = $id[$field];
+
```
Comment (phansys) [inline lib/Doctrine/ORM/Utility/IdentifierFlattener.php]: Please, remove surplus line break.

**Review by phansys — CHANGES_REQUESTED** [review]
Status: Needs work.

**Comment by Ocramius** [comment]
Please re-send this against master. Also, tests are absolutely needed, as well as an example of where this is happening.

File: lib/Doctrine/ORM/Utility/IdentifierFlattener.php (hunk 1 above)
Comment (gadelkareem) [inline lib/Doctrine/ORM/Utility/IdentifierFlattener.php]: Yes $id[$field] is null if Eager loading is not enabled.

**Comment by mpdude** [comment]
@gadelkareem is this still an issue?

**Pull request closed without merging**